src/components/GlobalChat.js

```diff
@@ -1,10 +1,41 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import OddTextInput from './oddx/OddTextInput';
 import 'stylesheets/GlobalChat.scss';
 
+import Service from 'services';
+
 const GlobalChat = () => {
+  const [randomID] = useState(Math.floor(Math.random() * 900000) + 100000);
+  const [ws, setWs] = useState(null);
+  const [message, setMessage] = useState([]);
+
+  useEffect(() => {
+    getChats().then(chats => setMessage(chats));
+
+    if (window['WebSocket']) {
+      const wsProtocol = window.location.protocol !== 'https:' ? 'ws://' : 'wss://';
+      const conn = new WebSocket(wsProtocol + document.location.host + '/ws');
+      setWs(conn);
+
+      conn.onclose = () => {
+        // Connection closed
+      };
+      conn.onmessage = e => {
+        const newMessages = e.data.split('\n');
+        setMessage(oldMessage => [...oldMessage, ...newMessages]);
+      };
+    }
+  }, []);
+
+  const getChats = async () => {
+    const data = await Service.getChat();
+    return data.map(e => e.content);
+  };
```

Comment:
lol, instead of calling from service directly, can we use context & reducer?

Reply:
Will integrate to redux later